From 37f2b71e03baa80f6af3d6b8fdb9fa0424418ca5 Mon Sep 17 00:00:00 2001
Message-Id: <37f2b71e03baa80f6af3d6b8fdb9fa0424418ca5.1687302523.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1687302523.git.sharlatanus@gmail.com>
References: <cover.1687302523.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 18 Jun 2023 12:38:10 +0100
Subject: [PATCH 01/20] gnu: sextractor: Update to 2.28.0, fix build.

* gnu/packages/astronomy.scm (sextractor): Update to 2.28.0, fix build.
[arguments]: Use G-expressions, add configure flags for CFitsio
detection.
[inputs]: Remove labels. Add cfitsio.
---
 gnu/packages/astronomy.scm | 34 +++++++++++++++++++---------------
 1 file changed, 19 insertions(+), 15 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index cdb6750f09..787ca01491 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -747,7 +747,7 @@ (define-public phd2
 (define-public sextractor
   (package
     (name "sextractor")
-    (version "2.25.0")
+    (version "2.28.0")
     (source
      (origin
        (method git-fetch)
@@ -756,29 +756,33 @@ (define-public sextractor
              (commit version)))
        (file-name (git-file-name name version))
        (sha256
-        (base32 "0q69n3nyal57h3ik2xirwzrxzljrwy9ivwraxzv9566vi3n4z5mw"))))
+        (base32 "15v7brjiraj2rdyxiidcgb58b3dqzdd363j31cjrfqhd1wc8ii5j"))))
     (build-system gnu-build-system)
     ;; NOTE: (Sharlatan-20210124T103117+0000): Building with `atlas' is failing
     ;; due to missing shared library which required on configure phase. Switch
     ;; build to use `openblas' instead. It requires FFTW with single precision
     ;; `fftwf'.
     (arguments
-     `(#:configure-flags
-       (list
-        "--enable-openblas"
-        (string-append
-         "--with-openblas-libdir=" (assoc-ref %build-inputs "openblas") "/lib")
-        (string-append
-         "--with-openblas-incdir=" (assoc-ref %build-inputs "openblas") "/include")
-        (string-append
-         "--with-fftw-libdir=" (assoc-ref %build-inputs "fftw") "/lib")
-        (string-append
-         "--with-fftw-incdir=" (assoc-ref %build-inputs "fftw") "/include"))))
+     (list
+      #:configure-flags
+      #~(list
+         "--enable-openblas"
+         (string-append "--with-cfitsio-incdir="
+                        #$(this-package-input "cfitsio") "/include")
+         (string-append "--with-cfitsio-libdir="
+                        #$(this-package-input "cfitsio") "/lib")
+         (string-append "--with-fftw-incdir="
+                        #$(this-package-input "fftwf") "/include")
+         (string-append "--with-fftw-libdir="
+                        #$(this-package-input "fftwf") "/lib")
+         (string-append "--with-openblas-incdir="
+                        #$(this-package-input "openblas") "/include")
+         (string-append "--with-openblas-libdir="
+                        #$(this-package-input "openblas") "/lib"))))
     (native-inputs
      (list autoconf automake libtool))
     (inputs
-     `(("openblas" ,openblas)
-       ("fftw" ,fftwf)))
+     (list cfitsio fftwf openblas))
     (home-page "https://www.astromatic.net/software/sextractor")
     (synopsis "Extract catalogs of sources from astronomical images")
     (description
-- 
2.40.1

